### PR TITLE
DSDEEPB-1244 Capture of return code on JES.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: src/main/scala/cromwell/engine/backend/jes/JesBackendCall.scala [hunk 1]
```diff
@@ -1,20 +1,57 @@
...
                           locallyQualifiedInputs: CallInputs,
                           callAbortRegistrationFunction: AbortRegistrationFunction) extends BackendCall {
+
+  import JesBackendCall._
+
+  def jesCommandLine = s"/bin/bash exec.sh > $StdoutFilename 2> $StderrFilename"
+
+  def downloadRcFile: String = {
+    val path = GoogleCloudStoragePath(rcJesOutput.gcs)
+    new String(backend.JesConnection.storage.downloadObject(path))
```
Comment (Horneth) [inline src/main/scala/cromwell/engine/backend/jes/JesBackendCall.scala]: There is a `slurpFile` method for that in storage I think

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 2]
```diff
@@ -271,16 +264,23 @@ class JesBackend extends Backend with LazyLogging {
...
+          FailedExecution(new Throwable(s"Workflow $workflowId: stderr has length $stderrLength for command: $command"))
+        case Run.Success if tryReturnCode.isFailure =>
+          FailedExecution(new Throwable(s"Workflow $workflowId: failed to download or parse return code file", tryReturnCode.failed.get))
+        case Run.Success if continueOnReturnCode.continueFor(tryReturnCode.get) =>
           unwrapOutputValues(outputMappings, backendCall.workflowDescriptor) match {
             case Success(outputs) => SuccessfulExecution(outputs)
             case Failure(e) => FailedExecution(e)
           }
+        case Run.Success =>
+          FailedExecution(new Throwable(s"Workflow $workflowId: disallowed command return code: " + tryReturnCode.get))
```
Comment (Horneth) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: This is a bit counter intuitive but not much we can do I guess.
Maybe fail on `case Run.Success if !continueOnReturnCode.continueFor(tryReturnCode.get)` above instead and finish with a `case Run.Success` which actually yields a SuccessfulExecution (after unwrapping)

Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: Yes, I had it that way originally but I thought we had a preference for doing it more like this?  I'd personally prefer to do it the way you suggest. :smile:

File: build.sbt [hunk 3]
```diff
@@ -3,7 +3,7 @@ import sbtrelease.ReleasePlugin._
 
 name := "cromwell"
 
-version := "0.12"
+version := "0.13"
```
Comment (mcovarr) [inline build.sbt]: No idea why this is showing up on my branch.

Comment (geoffjentry) [inline build.sbt]: I almost gave you your favorite "stowaway?" but figured it was intentional ;)

Comment (Horneth) [inline build.sbt]: From the commit history it looks like @scottfrazer committed it

**Comment by Horneth** [comment]
:+1: from me

File: build.sbt (hunk 3 above)
Comment (scottfrazer) [inline build.sbt]: Yes, I committed 0.13 directly to 'develop' after the release

File: src/main/scala/cromwell/engine/backend/jes/JesBackend.scala [hunk 4]
```diff
@@ -295,14 +292,18 @@ class JesBackend extends Backend with LazyLogging {
...
+  private def handleSuccess(outputMappings: Map[String, Try[WdlValue]], workflowDescriptor: WorkflowDescriptor): ExecutionResult = {
     val taskOutputEvaluationFailures = outputMappings filter { _._2.isFailure }
-    if (taskOutputEvaluationFailures.isEmpty) {
+    val outputValues = if (taskOutputEvaluationFailures.isEmpty) {
       Success(outputMappings collect { case (name, Success(wdlValue)) => name -> wdlValue })
     } else {
       val message = taskOutputEvaluationFailures collect { case (name, Failure(e)) => s"$name: $e" } mkString "\n"
       Failure(new Throwable(s"Workflow ${workflowDescriptor.id}: $message"))
     }
+    outputValues match {
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/jes/JesBackend.scala]: \n

File: src/main/scala/cromwell/engine/backend/jes/Run.scala [hunk 5]
```diff
@@ -11,14 +11,16 @@ import org.slf4j.LoggerFactory
...
 import scala.language.postfixOps
 
+
 object Run  {
   val JesServiceAccount = new ServiceAccount().setEmail("default").setScopes(GoogleScopes.Scopes.asJava)
   lazy val Log = LoggerFactory.getLogger("main")
-  lazy val maximumPollingInterval = ConfigFactory.load.getConfig("backend").getConfig("jes").getInt("maximumPollingInterval") * 1000
-  val initialPollingInterval = 500
-  val pollingBackoffFactor = 1.1
+  lazy val MaximumPollingInterval = ConfigFactory.load.getConfig("backend").getConfig("jes").getInt("maximumPollingInterval") * 1000
```
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/backend/jes/Run.scala]: I just made these changes on another branch :) great minds!

**Comment by geoffjentry** [comment]
:shipit:

File: build.sbt (hunk 3 above)
Comment (mcovarr) [inline build.sbt]: Argh the real problem is this PR is against master and not develop.  :frowning:  Closing and reopening...

**Pull request closed without merging**